tools/ci_connector_ops/ci_connector_ops/utils.py

```diff
@@ -74,6 +75,23 @@ def get_changed_acceptance_test_config(diff_regex: Optional[str] = None) -> Set[
     return {Connector(get_connector_name_from_path(changed_file)) for changed_file in changed_acceptance_test_config_paths}
 
 
+def get_gradle_project_dependencies_paths(gradle_file_path: Path):
+    dependencies_paths = []
+    with open(gradle_file_path) as gradle_file:
+        for line in gradle_file:
+            if "implementation project" in line or "implementation files" in line:
+                match = re.search(GRADLE_PROJECT_RE_PATTERN, line)
+                if match:
+                    project_name = match.group(2)
+                    if project_name.startswith(":"):
+                        project_path = project_name[1:].replace(":", "/")
+                        dependencies_paths.append(Path(project_path))
+                        dependency_gradle_file = Path(f"{project_path}/build.gradle")
```

Comment:
Oh man, such a shame that gradle doesn't have a good way to list all downstream dependencies 😢

Reply:
I could run `gradle dependencies` command inside a gradle container and parse its output, but it'll add a lot of overhead to this "pre pipeline run".